showcase-v1alpha2/src/v1alpha2/echo_client.ts

```diff
@@ -0,0 +1,688 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// TODO: streaming methods return untyped EventEmitter.
+// We cannot make stream.on() stronly typed because we return different types on
+// different events (e.g. stream.on('data') vs stream.on('metadata')). It may be
+// possible to list all valid types for on().
+
+import * as gax from 'google-gax';
+const gapicConfig = require('./echo_client_config') as gax.ClientConfig;
+import merge = require('lodash.merge');
+import * as path from 'path';
+
+const VERSION = require('../../package.json').version;
+
+export interface GapicClientOptions extends gax.GrpcClientOptions,
+                                            gax.GoogleAuthOptions,
+                                            gax.ClientStubOptions {
+  libName?: string;
+  libVersion?: string;
+  clientConfig?: gax.ClientConfig;
+}
+
+export interface Descriptors {
+  page: {[name: string]: gax.PageDescriptor};
+  stream: {[name: string]: gax.StreamDescriptor};
+}
+
+export interface GapicCallback<ResponseObject> {
+  (err?: Error|null, value?: ResponseObject|null): void;
+}
+
+export interface GapicPaginationCallback<
+    RequestObject, ResponseObject, ResponseType> {
+  (err?: Error|null, values?: ResponseType[], nextPageRequest?: RequestObject,
+   rawResponse?: ResponseObject): void;
+}
+
+export interface GapicPaginationResponse<
+    RequestObject, ResponseObject, ResponseType> {
+  values?: ResponseType[];
+  nextPageRequest?: RequestObject;
+  rawResponse?: ResponseObject;
+}
+
+// Interfaces for the service
+export interface Protos extends gax.GrpcObject {
+  google: Google;
+}
+
+export interface Google extends gax.GrpcObject {
+  showcase: GoogleShowcase;
+}
+
+export interface GoogleShowcase extends gax.GrpcObject {
+  v1alpha2: GoogleShowcaseV1alpha2;
+}
+
+export interface GoogleShowcaseV1alpha2 extends gax.GrpcObject {
+  Echo: typeof gax.ClientStub;
+}
+
+export interface EchoRequest {
+  content?: string;
+  error?: object;  // TODO: google.rpc.Status
+}
+
+export interface EchoResponse {
+  content?: string;
+}
+
+export interface ExpandRequest {
+  content?: string;
+  error?: object;  // TODO: google.rpc.Status
+}
+
+export interface WaitRequest {
+  response_delay: object;  // TODO: google.protobuf.Duration
+  error?: object;          // TODO: google.rpc.Status
+  success?: WaitResponse;
+}
+
+export interface WaitResponse {
+  content?: string;
+}
+
+export interface PaginationRequest {
+  max_response: number;
+  page_size?: number;
+  page_token?: string;
+}
+
+export interface PaginationResponse {
+  responses?: number[];
+  next_page_token?: string;
+}
+// End of interfaces
+
+export class EchoClient {
+  _descriptors: Descriptors = {page: {}, stream: {}};
```

Comment:
What is the expected behavior on private members? Are they supposed to be included? (Honestly asking; I trust you.)

Reply:
Hmmm. We can hide them, but I don't really see any benefits of doing that.  (not sure I answered the right question though - we can talk later)